rest-service/manager_rest/rest/resources_v1/executions.py

```diff
@@ -201,6 +201,58 @@ def post(self, **kwargs):
                 execution.id = execution_id
             sm.put(execution)
             messages = []
+            if (
+                force_status in ExecutionState.STATES
+                and force_status not in ExecutionState.WAITING_STATES
+                and force_status != ExecutionState.PENDING
+            ):
+                if deployment_id:
+                    deployment = sm.get(models.Deployment, deployment_id)
+                    deployment_latest_time = None
+                    if deployment.latest_execution is not None:
+                        deployment_latest_time = parse_datetime_string(
+                            deployment.latest_execution.started_at)
+
+                    exec_relations_set = False
+                    if (
+                        deployment_latest_time is None
+                        or (
+                            deployment_latest_time
+                            and deployment_latest_time < started_at
+                        )
+                    ):
```

Comment:
we do this stuff multiple times, maybe we can factor out a function here, something like

```python
def _is_execution_newer(started_at, execution):
    if not execution:
        return True
    x = parse(...)

# and then
if _is_execution_newer(started_at, dep.latest_execution):
    dep.latest_execution = execution
```

And this could be applied to latest execution, create execution, and blueprint upload execution

Reply:
Factored out.